my-app/src/App.js

```diff
@@ -0,0 +1,62 @@
+import React from "react";
+import logo from "./logo.svg";
+import "./reset.css";
+import "./App.css";
+import Nav from "./components/Nav.js";
+import Post from "./components/Post.js";
+import Posts from "./components/Posts.js";
+import More from "./components/More.js";
+
+class App extends React.Component {
+  constructor() {
+    super();
+    this.state = { posts: [], currentPage: 0 };
+  }
+
+  moreBtnClickHandler = e => {
+    this.setState(currentState => {
+      return {
+        posts: currentState.posts,
+        currentPage: ++currentState.currentPage
+      };
+    });
+  };
+
+  componentDidMount() {
+    this.fetchTopStories();
+  }
+
+  fetchTopStories() {
+    fetch("https://hacker-news.firebaseio.com/v0/topstories.json")
+      .then(response => response.json())
+      .then(data => {
+        data.map(storyId => {
+          fetch(` https://hacker-news.firebaseio.com/v0/item/${storyId}.json`)
+            .then(response => response.json())
+            .then(storyDetail => {
+              this.setState(currentState => {
+                currentState.posts.push(storyDetail);
+                return { posts: currentState.posts };
+              });
+            });
+        });
+      });
+  }
+  render() {
+    return (
+      <div className="App">
+        <Nav />
+        <Posts
+          currentPage={this.state.currentPage}
+          posts={this.state.posts.slice(
+            this.state.currentPage * 30,
+            (this.state.currentPage + 1) * 30
+          )}
```

Comment:
Try to avoid putting too much logic in the attributes themselves. For example, you could slice the array inside the `render` function, before you return your component. Then just use the result of that as the attribute.